src/character/subclass.rs

```diff
@@ -0,0 +1,118 @@
+
+/// XP can be earned for one subclass at a time whichever one is selected as the default
+/// Bonuses for each subclass can be applied together, for example you can be a level 10 hero with
+/// a level 5 subclass warrior, level 2 wizard and level cleric
+///
+/// TODO
+/// - implement talents and skills for subclasses to further increase play diversity
+
+pub enum SubclassType {
+    Warrior,
```

Comment:
couldn't this work as regular classes instead of subclasses? like, you would be a Warrior by default instead of a generic Hero, and then you can switch to other classes by command like you did

Reply:
I'm working on a bit of a write up to explain my idea for this, we can have a group discussion on whether this is the direction we want to go with or not in a bit.